src/nvim/buffer.c

```diff
@@ -3062,25 +3040,26 @@ build_stl_str_hl (
       s++;
       continue;
     }
-    opt = *s++;
+    char_u opt = *s++;
 
     /* OK - now for the real work */
-    base = 'D';
-    itemisflag = FALSE;
-    fillable = TRUE;
-    num = -1;
-    str = NULL;
+    char_u base = 'D';
+    bool itemisflag = false;
+    bool fillable = true;
+    long num = -1;
+    char_u *str = NULL;
     switch (opt) {
     case STL_FILEPATH:
     case STL_FULLPATH:
     case STL_FILENAME:
-      fillable = FALSE;         /* don't change ' ' to fillchar */
-      if (buf_spname(wp->w_buffer) != NULL)
+    {
+      fillable = false;         /* don't change ' ' to fillchar */
+      if (buf_spname(wp->w_buffer) != NULL) {
         STRLCPY(NameBuff, buf_spname(wp->w_buffer), MAXPATHL);
-      else {
-        t = (opt == STL_FULLPATH) ? wp->w_buffer->b_ffname
-            : wp->w_buffer->b_fname;
-        home_replace(wp->w_buffer, t, NameBuff, MAXPATHL, TRUE);
```

Comment:
the linter told me to change `TRUE` to `true` on this line, but I'm not sure why it complained because I didn't change this line and it was fine before?